File: csharp/Slimmed/PublishingHouse.Api/Requests/CreateDraftRequest.cs
namespace PublishingHouse.Api.Requests;

public record CreateDraftRequest(
    string? Title,
    AuthorRequest? Author,
    Guid? PublisherId,
    int? Edition,
    string? Genre
);

public record AuthorRequest(Guid? AuthorId, string? FirstName, string? LastName);

File: java/src/main/java/io/eventdriven/slimdownaggregates/original/api/requests/CreateDraftRequest.java
package io.eventdriven.slimdownaggregates.original.api.requests;

import java.util.UUID;

public record CreateDraftRequest(
  String title,
  AuthorRequest author,
  UUID publisherId,
  Integer edition,
  String genre
) {
  public record AuthorRequest(UUID authorId, String firstName, String lastName) {
  }
}
